public/app/plugins/datasource/cloud-monitoring/components/Experimental/AlignmentPeriodLabel.tsx

```diff
@@ -0,0 +1,38 @@
+import { css } from '@emotion/css';
+import React, { FC, useMemo } from 'react';
+
+import { rangeUtil, GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+import { ALIGNMENTS } from '../../constants';
+import CloudMonitoringDatasource from '../../datasource';
+import { CustomMetaData } from '../../types';
+
+export interface Props {
+  customMetaData: CustomMetaData;
+  datasource: CloudMonitoringDatasource;
+}
+
+export const AlignmentPeriodLabel: FC<Props> = ({ customMetaData, datasource }) => {
+  const { perSeriesAligner, alignmentPeriod } = customMetaData;
+  const styles = useStyles2(getStyles);
+  const formatAlignmentText = useMemo(() => {
+    if (!alignmentPeriod || !perSeriesAligner) {
+      return '';
+    }
+
+    const alignment = ALIGNMENTS.find((ap) => ap.value === datasource.templateSrv.replace(perSeriesAligner));
+    const seconds = parseInt(alignmentPeriod ?? ''.replace(/[^0-9]/g, ''), 10);
```

Comment:
I think this could be simplified to: `parseInt(alignmentPeriod, 10)` since we're checking up on line 20 already.